airdrop/job/eligibility.py

```diff
@@ -0,0 +1,95 @@
+
+import sys
+sys.path.append('/opt')
+import time
+import uuid
+
+from airdrop.job.repository import Repository
+from common.exception_handler import exception_handler
+from common.utils import generate_lambda_response
+from airdrop.config import BALANCE_DB_CONFIG, NETWORK, SLACK_HOOK
+from common.logger import get_logger
+from pydoc import locate
+from decimal import Decimal
+
+logger = get_logger(__name__)
+
+class EligibilityProcessor:
+    def __init__(self):
+        self._airdrop_db = Repository(NETWORK["db"])
+        self._balances_db = Repository(BALANCE_DB_CONFIG)
+        #self._airdrop_windows_open_for_snapshot = []
+        self._active_airdrop_window_map = {}
+        self._snapshot_guid = str(uuid.uuid4())
+        self.__insert_snapshot = "insert into user_balance_snapshot (airdrop_window_id, address, balance, staked, total, snapshot_guid, row_created, row_updated) "+\
+                          "values(%s,%s,%s,%s,%s, %s, current_timestamp, current_timestamp)"
+        self.__rows_to_insert = []
+        return
+
+    def __populate_state(self):
+        all_windows = self._airdrop_db.execute("select aw.airdrop_id, ad.rewards_processor, aw.row_id as airdrop_window_id from airdrop_window aw, airdrop ad " + \
+                                               "where aw.airdrop_id = ad.row_id and ad.rewards_processor is not null " + \
+                                               "and current_timestamp between aw.first_snapshot_at and aw.last_snapshot_at ")
+        for window in all_windows:
+            details = {}
+            details["airdrop_id"] = window["airdrop_id"]
+            details["rewards_processor"] = window["rewards_processor"]
+            self._active_airdrop_window_map[window["airdrop_window_id"]] = details
+        
+    def __batch_insert(self, values, force=False):
+        start = time.process_time()
+        number_of_rows = len(self.__rows_to_insert)
+        if (force and number_of_rows > 0) or number_of_rows >= 50:
+            self._airdrop_db.bulk_query(self.__insert_snapshot, self.__rows_to_insert)
+            self.__rows_to_insert.clear()       
+            print(f"*****{(time.process_time() - start)} seconds. Inserted {number_of_rows} rows")
```

Comment:
We should make use of logger instead of print.

Reply:
Yeah, this is incomplete. Will clean this up when I complete the logic